From: Javad Rahmatzadeh <j.rahmatzadeh@gmail.com>
Date: Mon, 10 Aug 2026 21:04:32 +0200
Subject: qrCode: Rename url property to text

The `QrCode` class supports generating QR codes for any string,
not only URLs.

Rename the property because technically it can be used for strings
other than the URLs it was intended for.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/4329>

Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/00ebceb
---
 js/gdm/webLogin.js | 2 +-
 js/ui/qrCode.js    | 8 ++++----
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/js/gdm/webLogin.js b/js/gdm/webLogin.js
index 3b64962..62597b4 100644
--- a/js/gdm/webLogin.js
+++ b/js/gdm/webLogin.js
@@ -85,7 +85,7 @@ export class WebLoginPrompt extends St.BoxLayout {
             this._urlTitleLabel.text = message;
 
         if (url !== null) {
-            this._qrCode.set({url});
+            this._qrCode.set({text: url});
 
             const formattedUrl = this._formatURLForDisplay(url);
             this._urlLabel.text = formattedUrl;
diff --git a/js/ui/qrCode.js b/js/ui/qrCode.js
index caf0999..4a68753 100644
--- a/js/ui/qrCode.js
+++ b/js/ui/qrCode.js
@@ -13,8 +13,8 @@ const QR_CODE_TRANSPARENT_COLOR = new GnomeQR.Color({alpha: 0});
 
 export class QrCode extends St.Bin {
     static [GObject.properties] = {
-        'url': GObject.ParamSpec.string(
-            'url', null, null,
+        'text': GObject.ParamSpec.string(
+            'text', null, null,
             GObject.ParamFlags.READWRITE,
             null),
     };
@@ -45,7 +45,7 @@ export class QrCode extends St.Bin {
 
         this._fgColor = null;
 
-        this.connect('notify::url', () =>
+        this.connect('notify::text', () =>
             this._update().catch(logErrorUnlessCancelled));
     };
 
@@ -80,7 +80,7 @@ export class QrCode extends St.Bin {
         this._cancellable = cancellable;
 
         const [pixelData, qrSize] = await GnomeQR.generate_qr_code_async(
-            this.url,
+            this.text,
             0 /* size, are fine with the minimum value */,
             QR_CODE_TRANSPARENT_COLOR,
             fgColor,
